Diagnostic/Providers/Builtin.py

```diff
@@ -0,0 +1,183 @@
+        self._Label = ProvUtil.GetCounterSetting(counterSpec, 'counterSpecifier')
+        self._SampleRate = ProvUtil.GetCounterSetting(counterSpec, 'sampleRate')
+
+    def IsType(self, t):
+        return self.Type == t.lower()
+
+    def Class(self):
+        return self._CounterClass
+
+    def Counter(self):
+        return self._Counter
+
+    def InstanceId(self):
+        return self._InstanceId
+
+    def Label(self):
+        return self._Label
+
+    def SampleRate(self):
+        if self._SampleRate is None:
+            return _defaultSampleRate
+        else:
+            return ProvUtil.IntervalToSeconds(self._SampleRate)
+
+
+def AddMetric(counterSpec):
+    try:
+        metric = BuiltinMetric(counterSpec)
+    except ProvUtil.ParseException as ex:
+        print "Couldn't create metric: ", ex
+        return
+
+    # (class, instanceId, sampleRate) -> [ metric ]
+    # Given a class, instance within that class, and sample rate, we have a list of the requested metrics
+    # matching those constraints. For that set of constraints, we also have a common eventName, the local
+    # table where we store the collected metrics.
+
+    key = (metric.Class(), metric.InstanceId(), metric.SampleRate() )
+    if (key not in _eventNames):
+        _eventNames[key] = ProvUtil.MakeUniqueEventName('builtin')
+    _metrics[key].append(metric)
+
+def GenerateOMIQuery():
+    queries = []
+    for group in _metrics:
+        (className, instanceId, sampleRate) = group
+        if className in _instancedClasses and not instanceId:
+            instanceId = '_Total'
+        columns = []
+        mappings = []
+        for metric in _metrics[group]:
+            columns.append(metric.Counter())
+            mappings.append('<MapName name="{0}">{1}</MapName>'.format(metric.Counter(), metric.Label()))
+        columnString = ','.join(columns)
+        if instanceId:
+            whereClause = " WHERE name='{0}'".format(instanceId)
+        else:
+            whereClause = ""
+        queries.append('''
+<OMIQuery cqlQuery="SELECT {0} FROM {1}{2}" eventName="{3}" omiNamespace="root/scx" sampleRateInSeconds="{4}" storeType="local">
+  <Unpivot columnName="CounterName" columnValue="Value" columns="{0}">
+    {5}
+  </Unpivot>
+</OMIQuery>'''.format(
+            columnString,
+            _omiClassName[className],
+            whereClause,
+            _eventNames[group],
+            sampleRate,
+            '\n    '.join(mappings)
+        ))
+    return ''.join(queries)
+
+
+#import base64
+#import datetime
+#import os
+#import os.path
+#import platform
+#import re
+#import signal
+#import string
+#import subprocess
+#import sys
+#import syslog
+#import time
+#import traceback
+#import xml.dom.minidom
+#import xml.etree.ElementTree as ET
+#
+#import Utils.HandlerUtil as Util
+#import Utils.LadDiagnosticUtil as LadUtil
+#import Utils.XmlUtil as XmlUtil
+#import Utils.ApplicationInsightsUtil as AIUtil
+#from Utils.WAAgentUtil import waagent
```

Comment:
As noted in the introductory comments to the PR, this module is not active as a result of this PR, but I wanted to get it reviewed sooner rather than later. Unit tests are still required before the code would be ready for integration. I've added a significant comment at the beginning of the Builtin.py provider which attempts to explain how all this works.